test/stdlib/StringMemoryTest.swift

```diff
@@ -0,0 +1,66 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -O %s -o %t/StringMemoryTest
+// RUN: %target-run %t/StringMemoryTest | %FileCheck %s
+
+// REQUIRES: optimized_stdlib
+// REQUIRES: executable_test
+// REQUIRES: objc_interop
+
+// The autorelease return optimizaion seems to be failing in the simulator.
+// XFAIL: CPU=i386
```

Comment:
I think we just don't do it there. @rjmccall?

Reply:
Yes, seems to be the case. I tried with a pure objective-c app and we also leak there on an i386 simulator rdar://34863458